src/test/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/SurveyLaunchedEndpointUnitTest.java

```diff
@@ -40,56 +46,57 @@ public void setUp() throws Exception {
             .setHandlerExceptionResolvers(mockAdviceFor(RestExceptionHandler.class))
             .setMessageConverters(new MappingJackson2HttpMessageConverter(new CustomObjectMapper()))
             .build();
+    dto = FixtureHelper.loadClassFixtures(SurveyLaunchedDTO[].class).get(0);
   }
 
   @Test
   public void surveyLaunchedSuccessCaseEmptyString() throws Exception {
     Mockito.doNothing().when(surveyLaunchedService).surveyLaunched(any());
-
-    String surveyLaunchedRequestBody =
-        "{ \"questionnaireId\": \"23434234234\", "
-            + "\"caseId\": \"3fa85f64-5717-4562-b3fc-2c963f66afa6\", "
-            + "\"agentId\": \"\"}";
-    mockMvc
-        .perform(postJson("/surveyLaunched", surveyLaunchedRequestBody))
-        .andExpect(status().isOk());
+    callEndointExpectingSuccess();
   }
 
   @Test
   public void surveyLaunchedSuccessCaseAssistedDigitalLocation() throws Exception {
     Mockito.doNothing().when(surveyLaunchedService).surveyLaunched(any());
+    callEndointExpectingSuccess();
+  }
 
-    String surveyLaunchedRequestBody =
-        "{ \"questionnaireId\": \"23434234234\", "
-            + "\"caseId\": \"3fa85f64-5717-4562-b3fc-2c963f66afa6\", "
-            + "\"agentId\": \"1000007\"}";
-    mockMvc
-        .perform(postJson("/surveyLaunched", surveyLaunchedRequestBody))
-        .andExpect(status().isOk());
+  @Test
+  public void shouldRejectMissingQuestionnaireId() throws Exception {
+    dto.setQuestionnaireId(null);
+    callEndointExpectingBadRequest();
   }
 
   @Test
-  public void surveyLaunchedWithMissingQuestionnaireId() throws Exception {
-    String surveyLaunchedRequestBody = "{ \"caseId\": \"3fa85f64-5717-4562-b3fc-2c963f66afa6\" }";
-    mockMvc
-        .perform(postJson("/surveyLaunched", surveyLaunchedRequestBody))
-        .andExpect(status().isBadRequest());
+  public void shouldLaunchWithMissingClientIP() throws Exception {
+    dto.setClientIP(null);
+    callEndointExpectingSuccess();
   }
 
   @Test
-  public void surveyLaunchedWithInvalidCaseId() throws Exception {
+  public void shouldRejectInvalidCaseId() throws Exception {
     String surveyLaunchedRequestBody =
         "{ \"questionnaireId\": \"23434234234\",   \"caseId\": \"euieuieu@#$@#$\" }";
-    mockMvc
-        .perform(postJson("/surveyLaunched", surveyLaunchedRequestBody))
-        .andExpect(status().isBadRequest());
+    callEndointExpectingBadRequest(surveyLaunchedRequestBody);
   }
 
   @Test
-  public void surveyLaunchedWithInvalidRequest() throws Exception {
+  public void shouldRejectInvalidRequest() throws Exception {
     String surveyLaunchedRequestBody = "uoeuoeu 45345345 euieuiaooo";
+    callEndointExpectingBadRequest(surveyLaunchedRequestBody);
+  }
+
+  private void callEndointExpectingSuccess() throws Exception {
```

Comment:
This looks like a typo
```suggestion
  private void callEndpointExpectingSuccess() throws Exception {
```

Reply:
thanks Ella, corrected